0.12 cty panic: value has no attribute of that name During Import

### Terraform Version

```
github.com/hashicorp/terraform v0.12.0-alpha4.0.20190201001850-411dab1dea1a
github.com/zclconf/go-cty v0.0.0-20190130221141-d7fe3fa0020f
```

### Terraform Configuration Files

```hcl
resource "aws_vpc" "example" {
  cidr_block = "10.1.0.0/16"
}

resource "aws_security_group" "example" {
  name = "example"
  vpc_id = "${aws_vpc.example.id}"

  ingress {
    protocol = "6"
    from_port = 80
    to_port = 8000
    cidr_blocks = ["10.0.0.0/8"]
  }

  tags = {
    Name = "example"
  }
}
```

### Debug Output

OSS TeamCity Link: https://HOSTNAME/viewLog.html?buildId=43313&buildTypeId=Aws_ProviderAws&tab=artifacts

### Crash Output

```
=== CONT  TestAccAWSSecurityGroup_importBasic

------- Stderr: -------
panic: value has no attribute of that name

goroutine 984 [running]:
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/zclconf/go-cty/cty.Value.GetAttr(0x4b64760, 0xc0002b33f8, 0x3b4ab20, 0xc000818510, 0xc001b39294, 0xb, 0x0, 0x1, 0xc0003a9860, 0xc0003a9b30)
	/opt/teamcity-agent/work/2e10e023da0c7520/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/zclconf/go-cty/cty/value_ops.go:586 +0x296
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/zclconf/go-cty/cty/msgpack.marshal(0x4b64760, 0xc0002b33f8, 0x3b4ab20, 0xc000818510, 0x4b64760, 0xc0002b3410, 0x0, 0x0, 0x0, 0xc00196a8d0, ...)
	/opt/teamcity-agent/work/2e10e023da0c7520/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/zclconf/go-cty/cty/msgpack/marshal.go:176 +0x488
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/zclconf/go-cty/cty/msgpack.Marshal(0x4b64760, 0xc0002b33f8, 0x3b4ab20, 0xc000818510, 0x4b64760, 0xc0002b3410, 0x0, 0x4d, 0xc000709d60, 0x17, ...)
	/opt/teamcity-agent/work/2e10e023da0c7520/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/zclconf/go-cty/cty/msgpack/marshal.go:35 +0x21c
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/plugin.(*GRPCProvider).ReadResource(0xc000892080, 0xc000709d60, 0x17, 0x4b64760, 0xc0002b33f8, 0x3b4ab20, 0xc000818510, 0x0, 0x0, 0x0, ...)
	/opt/teamcity-agent/work/2e10e023da0c7520/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/plugin/grpc_provider.go:320 +0x10a
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/terraform.(*EvalRefresh).Eval(0xc000e5ae10, 0x4b8b340, 0xc000c3d1e0, 0x2, 0x2, 0x0, 0x0)
	/opt/teamcity-agent/work/2e10e023da0c7520/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/terraform/eval_refresh.go:61 +0x40e
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/terraform.EvalRaw(0x4b52280, 0xc000e5ae10, 0x4b8b340, 0xc000c3d1e0, 0x0, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/2e10e023da0c7520/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/terraform/eval.go:57 +0x167
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/terraform.(*EvalSequence).Eval(0xc0008f8f00, 0x4b8b340, 0xc000c3d1e0, 0x2, 0x2, 0x7ff818a2d568, 0x67)
	/opt/teamcity-agent/work/2e10e023da0c7520/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/terraform/eval_sequence.go:20 +0xfd
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/terraform.EvalRaw(0x4b522e0, 0xc0008f8f00, 0x4b8b340, 0xc000c3d1e0, 0x3b40560, 0x83c36a7, 0x3784940, 0xc002227de0)
	/opt/teamcity-agent/work/2e10e023da0c7520/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/terraform/eval.go:57 +0x167
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/terraform.Eval(0x4b522e0, 0xc0008f8f00, 0x4b8b340, 0xc000c3d1e0, 0xc0008f8f00, 0x4b522e0, 0xc0008f8f00, 0x0)
	/opt/teamcity-agent/work/2e10e023da0c7520/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/terraform/eval.go:35 +0x4d
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/terraform.(*Graph).walk.func1(0x3d84a60, 0xc0023a05b0, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/2e10e023da0c7520/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/terraform/graph.go:90 +0x10d1
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/dag.(*Walker).walkVertex(0xc000f85180, 0x3d84a60, 0xc0023a05b0, 0xc000d16380)
	/opt/teamcity-agent/work/2e10e023da0c7520/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/dag/walk.go:392 +0x373
created by github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/dag.(*Walker).Update
	/opt/teamcity-agent/work/2e10e023da0c7520/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/dag/walk.go:314 +0x9d2
```

### Expected Behavior

No crash. 😄 

### Actual Behavior

Crash. 😅 

### Steps to Reproduce

From an AWS provider branch that has 0.12 update, e.g. `v0.12-sdk-updates-411dab1dea1a`

```
$ TF_ACC=1 go test ./aws -v -timeout 120m -parallel 20 -run=TestAccAWSSecurityGroup_importBasic
```

### Additional Context

Its either related to import or is a strange coincidence to be seemingly only happen with import tests. This also seems to happen with `TestAccAwsDxGateway_importComplex`, so it may be related to or a continuation of the `ResourceData.SetType()` work done recently in #20148

```
=== CONT  TestAccAwsDxGateway_importComplex

------- Stderr: -------
panic: value has no attribute of that name

goroutine 2608 [running]:
github.com/terraform-providers/terraform-provider-aws/vendor/github.com/zclconf/go-cty/cty.Value.GetAttr(0x4b64760, 0xc000cf8aa0, 0x3b4ab20, 0xc001e5ede0, 0xc000e180e0, 0xd, 0x0, 0x1, 0x8, 0x0)
	/opt/teamcity-agent/work/2e10e023da0c7520/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/zclconf/go-cty/cty/value_ops.go:586 +0x296
```

### References

* Related/continuation of: #20148

--- a/builtin/providers/test/resource_import_other.go
+++ b/builtin/providers/test/resource_import_other.go
@@ -1,6 +1,8 @@
 package test
 
 import (
+	"fmt"
+
 	"github.com/hashicorp/terraform/helper/schema"
 )
 
@@ -26,23 +28,9 @@ func testResourceImportOther() *schema.Resource {
 				Optional: true,
 				Default:  true,
 			},
-			"nested": {
-				Type:     schema.TypeSet,
-				Optional: true,
-				ForceNew: true,
-				Elem: &schema.Resource{
-					Schema: map[string]*schema.Schema{
-						"string": {
-							Type:     schema.TypeString,
-							Optional: true,
-							Default:  "default nested",
-						},
-						"optional": {
-							Type:     schema.TypeString,
-							Optional: true,
-						},
-					},
-				},
+			"computed": {
+				Type:     schema.TypeString,
+				Computed: true,
 			},
 		},
 	}
@@ -74,10 +62,13 @@ func testResourceImportOtherUpdate(d *schema.ResourceData, meta interface{}) err
 }
 
 func testResourceImportOtherRead(d *schema.ResourceData, meta interface{}) error {
+	err := d.Set("computed", "hello!")
+	if err != nil {
+		return fmt.Errorf("failed to set 'computed' attribute: %s", err)
+	}
 	return nil
 }
 
 func testResourceImportOtherDelete(d *schema.ResourceData, meta interface{}) error {
-	d.SetId("")
 	return nil
 }

--- a/builtin/providers/test/resource_import_other_test.go
+++ b/builtin/providers/test/resource_import_other_test.go
@@ -38,6 +38,8 @@ resource "test_resource_import_other" "foo" {
 						return fmt.Errorf("no instance with id import_other_main in state")
 					} else if got, want := is.Ephemeral.Type, "test_resource_import_other"; got != want {
 						return fmt.Errorf("import_other_main has wrong type %q; want %q", got, want)
+					} else if got, want := is.Attributes["computed"], "hello!"; got != want {
+						return fmt.Errorf("import_other_main has wrong value %q for its computed attribute; want %q", got, want)
 					}
 					if is, ok := byID["import_other_other"]; !ok {
 						return fmt.Errorf("no instance with id import_other_other in state")

--- a/go.mod
+++ b/go.mod
@@ -128,7 +128,7 @@ require (
 	github.com/xanzy/ssh-agent v0.2.0
 	github.com/xiang90/probing v0.0.0-20160813154853-07dd2e8dfe18 // indirect
 	github.com/xlab/treeprint v0.0.0-20161029104018-1d6e34225557
-	github.com/zclconf/go-cty v0.0.0-20190130221141-d7fe3fa0020f
+	github.com/zclconf/go-cty v0.0.0-20190201220620-4ca19710f056
 	go.opencensus.io v0.17.0 // indirect
 	go.uber.org/atomic v1.3.2 // indirect
 	go.uber.org/multierr v1.1.0 // indirect

--- a/helper/plugin/grpc_provider.go
+++ b/helper/plugin/grpc_provider.go
@@ -794,8 +794,6 @@ func (s *GRPCProviderServer) ApplyResourceChange(_ context.Context, req *proto.A
 func (s *GRPCProviderServer) ImportResourceState(_ context.Context, req *proto.ImportResourceState_Request) (*proto.ImportResourceState_Response, error) {
 	resp := &proto.ImportResourceState_Response{}
 
-	block := s.getResourceSchemaBlock(req.TypeName)
-
 	info := &terraform.InstanceInfo{
 		Type: req.TypeName,
 	}
@@ -810,6 +808,12 @@ func (s *GRPCProviderServer) ImportResourceState(_ context.Context, req *proto.I
 		// copy the ID again just to be sure it wasn't missed
 		is.Attributes["id"] = is.ID
 
+		resourceType := is.Ephemeral.Type
+		if resourceType == "" {
+			resourceType = req.TypeName
+		}
+
+		block := s.getResourceSchemaBlock(resourceType)
 		newStateVal, err := hcl2shim.HCL2ValueFromFlatmap(is.Attributes, block.ImpliedType())
 		if err != nil {
 			resp.Diagnostics = convert.AppendProtoDiag(resp.Diagnostics, err)
@@ -826,11 +830,6 @@ func (s *GRPCProviderServer) ImportResourceState(_ context.Context, req *proto.I
 		if err != nil {
 			resp.Diagnostics = convert.AppendProtoDiag(resp.Diagnostics, err)
 			return resp, nil
-		}
-
-		resourceType := is.Ephemeral.Type
-		if resourceType == "" {
-			resourceType = req.TypeName
 		}
 
 		importedResource := &proto.ImportResourceState_ImportedResource{

--- a/plugin/grpc_provider.go
+++ b/plugin/grpc_provider.go
@@ -459,8 +459,6 @@ func (p *GRPCProvider) ApplyResourceChange(r providers.ApplyResourceChangeReques
 func (p *GRPCProvider) ImportResourceState(r providers.ImportResourceStateRequest) (resp providers.ImportResourceStateResponse) {
 	log.Printf("[TRACE] GRPCProvider: ImportResourceState")
 
-	resSchema := p.getResourceSchema(r.TypeName)
-
 	protoReq := &proto.ImportResourceState_Request{
 		TypeName: r.TypeName,
 		Id:       r.ID,
@@ -479,6 +477,7 @@ func (p *GRPCProvider) ImportResourceState(r providers.ImportResourceStateReques
 			Private:  imported.Private,
 		}
 
+		resSchema := p.getResourceSchema(resource.TypeName)
 		state := cty.NullVal(resSchema.Block.ImpliedType())
 		if imported.State != nil {
 			state, err = msgpack.Unmarshal(imported.State.Msgpack, resSchema.Block.ImpliedType())

